packages/theme/src/components/TsDocSignatureType.tsx

```diff
@@ -0,0 +1,226 @@
+import React from 'react';
+import TsDocSignatureTitle from './TsDocSignatureTitle';
+
+
+function Parens({
+  children,
+  needParens = false,
+  openParen = '(',
+  closeParen = ')'
+}: {
+  children: React.ReactNode;
+  needParens?: boolean;
+  openParen?: string;
+  closeParen?: string;
+}) {
+  return (
+    <>
+      {needParens && openParen}
+      {children}
+      {needParens && closeParen}
+    </>
+  );
+}
+
+interface Props {
+  type: any;
+  needParens?: boolean;
+}
+
+export default function TsDocSignatureType({
+  type,
+  needParens = false
+}: Props) {
+  if (!type) return '';
+
+  if (
+    type.type === 'intrinsic' ||
+    type.type === 'typeParameter' ||
+    type.type === 'unknown'
+  ) {
+    return type.name;
+  }
+
+  if (type.type === 'array') {
+    return (
+      <>
+        <TsDocSignatureType type={type.elementType} needParens />
+        []
+      </>
+    );
+  }
+
+  if (type.type === 'conditional') {
+    return (
+      <Parens needParens={needParens}>
+        <TsDocSignatureType type={type.checkType} needParens />
+        extends
+        <TsDocSignatureType type={type.extendsType} needParens />
+        ?
+        <TsDocSignatureType type={type.trueType} />
+        :
+        <TsDocSignatureType type={type.falseType} />
+      </Parens>
+    );
+  }
+
+  if (type.type === 'indexedAccess') {
+    return (
+      <>
+        <TsDocSignatureType type={type.objectType} />
+        [<TsDocSignatureType type={type.indexType} />]
+      </>
+    );
+  }
+
+  if (type.type === 'inferred') {
+    return `inger ${type.name}`;
+  }
+
+  if (type.type === 'intersection') {
+    return (
+      <Parens needParens={needParens}>
+        {type.types.map((tt, idx) => (
+          <React.Fragment key={tt.id}>
+            {idx > 0 &&' & '}
+            <TsDocSignatureType key={tt.id} type={tt} />
+          </React.Fragment>
+        ))}
+      </Parens>
+    );
+  }
+
+  if (type.type === 'predicate') {
+    return (
+      <>
+        {type.asserts && 'asserts '}
+        {type.name}
+        {type.targetType && <TsDocSignatureType type={type.targetType} />}
+      </>
+    );
+  }
+
+  if (type.type === 'query') {
+    return (
+      <>
+        typeof
+        <TsDocSignatureType type={type.queryType} />
+      </>
+    );
+  }
+
+  if (type.type === 'reference') {
+    return (
+      <>
+        {type.reflection ? type.reflection.name : type.name}
+        {type.typeArguments && (
+          <>
+            &lt;
+            {type.typeArguments.map((typeArgument, idx) => (
+              <React.Fragment key={typeArgument.id}>
+                {idx > 0 &&', '}
+                <TsDocSignatureType key={typeArgument.id} type={typeArgument} />
+              </React.Fragment>
+            ))}
+            &gt;
+          </>
+        )}
+      </>
+    );
+  }
+
+  if (type.type === 'reflection') {
+    if (type.declaration.children) {
```

Comment:
likely this should merge with `reference` above since they both point to a TypedocNode. 

Also there is no `children` that is a reserved field by Gatsby, so the field you actually want here is `typedocs`